corehq/apps/hqwebapp/forms.py

```diff
@@ -44,7 +44,7 @@ def clean(self):
         if not password:
             raise ValidationError(_("Please enter a password."))
 
-        if settings.ENABLE_DRACONIAN_SECURITY_FEATURES:
+        if 'captcha' in self.fields:
             if not self.cleaned_data.get('captcha'):
                 raise ValidationError(_("Please enter valid CAPTCHA"))
```

Comment:
With this change I think this validation could be bypassed on the client by simply not submitting a `captcha` field? That seems like a security hole.
```suggestion
        if settings.ADD_CAPTCHA_FIELD_TO_FORMS:
            if not self.cleaned_data.get('captcha'):
                raise ValidationError(_("Please enter valid CAPTCHA"))
```

Reply:
That is a great point. Tested that even if the user omits/deletes the field on the form in the UI or if the request does not have captcha data in form, 'captcha' is still in `self.fields` because we are not checking for data submitted but the fields of the form in code. So it is true today but I am happy to change it to the suggested implementation, which was also done originally, if it looks like this could get missed.

Reply:
Thanks for checking. I don't see a reason to introduce a new way of checking for `settings.ADD_CAPTCHA_FIELD_TO_FORMS`, so the suggestion seems slightly better.